File: dev/python_libs/bidder.py
from python_libs.loggers import logger
from python_libs import config
import random 

class Bidder:
    def __init__(self,id):
        self.id=id
        self.behaviour={}
        pass

    def set_behaviour(self,**kwargs):
        for key, value in kwargs.items(): 
            self.behaviour[key]=value

    def order(self):
        order_type=random.choice(['buy','sell']) if "order_type" not in self.behaviour else self.behaviour["order_type"]
        stop=random.randint(config.MINIMUM_STOP, config.MAXIMUM_STOP) if "stop" not in self.behaviour else self.behaviour["stop"]
        limit=random.randint(config.MINIMUM_LIMIT, config.MAXIMUM_LIMIT) if "limit" not in self.behaviour else self.behaviour["limit"] 
        return {
            "order_type":order_type,
            "stop":stop,
            "limit":limit
        }

File: nodejs-streamer/modules/bidder.js
const choose=(choices)=> {
    var index = Math.floor(Math.random() * choices.length);
    return choices[index];
  }

class Bidder{
    constructor(behaviour,config){
        this.DEFAULT_LIMIT=config.DEFAULT_LIMIT? config.DEFAULT_LIMIT:10
        this.DEFAULT_STOP=config.DEFAULT_STOP? config.DEFAULT_STOP:10
        this.behaviour=behaviour;
        // e.g. behaviour:
        // {   order_type: 'buy '}
    }

    order(){
        let order={
            type:("order_type" in this.behaviour)? this.behaviour.order_type:choose(['buy','sell']),
            stop:this.DEFAULT_STOP,
            limit:this.DEFAULT_LIMIT
        }

        return order
    }

}

module.exports=Bidder